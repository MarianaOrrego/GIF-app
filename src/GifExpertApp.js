import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGird } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    //const [categories, setCategories] = useState(['Animales']);
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <div className='contenedor'>
                <div className='title'>
                    <h1>GIF App</h1>
                </div>
                <div className='ubicacion'>
                    <AddCategory setCategories={setCategories} />
                    <hr />
                </div>
            </div>

            <ol>
                {
                    categories.map(category =>

                        <GifGird
                            category={category}
                            key={category}
                        />
                    )
                }
            </ol>
        </>
    );

};

