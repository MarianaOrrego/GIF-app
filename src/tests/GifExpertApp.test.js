import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en GifExpertApp', () => {
    
    test(' Debe mostrar el componenete correctamente ', () => {
        
        const wrapper = shallow ( < GifExpertApp />); 
        expect( wrapper ).toMatchSnapshot();
    });

    test(' Debe de mostrar una lista de categorias ', () => {
        
        const categories = ['Perro', 'Gato'];
        const wrapper = shallow( < GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGird').length).toBe( categories.length );

    });
    
    
})
