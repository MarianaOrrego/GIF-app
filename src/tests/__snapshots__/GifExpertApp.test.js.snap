// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp  Debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGird
      category="Perro"
      key="Perro"
    />
    <GifGird
      category="Gato"
      key="Gato"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp  Debe mostrar el componenete correctamente  1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
